### DISCUSSION TITLE
Why is the Docker container updating so often?

### DISCUSSION THREAD

**Discussion by mihawk90** [automatic-ripping-machine/automatic-ripping-machine#1241] [Q&A] [answered]
This is more of a curiosity then anything else but I only noticed it with the ARM container, so here goes.
I'm running ARM on TrueNAS Scale and their GUI has this nice feature of showing updates for installed Containers/Apps (similar to Watchtower). With ARM specifically I am noticing that there are updates available almost daily, but I can't quite figure out why.

Checking the [DockerHub Tags](https://hub.docker.com/r/automaticrippingmachine/automatic-ripping-machine/tags) I can see that the Docker tags are indeed being updated, so at least it's not a TrueNAS issue.

The current `latest` tag is
> Last pushed 6 hours ago by [automaticrippingmachine](https://hub.docker.com/u/automaticrippingmachine)
>  | Digest | OS/ARCH | Compressed Size |
>  |-----------|---------------|--------------------------|
>  | [30a5f129e72f](https://hub.docker.com/layers/automaticrippingmachine/automatic-ripping-machine/latest/images/sha256-30a5f129e72f7d40916ee4b4293dcaaef66f02f4f62b198070fb3618f09b5879?context=explore) | linux/amd64 | 853.58 MB |
>  | [5dcf7dd8ae2c](https://hub.docker.com/layers/automaticrippingmachine/automatic-ripping-machine/latest/images/sha256-5dcf7dd8ae2cf35e6309d5624d2ec41843f9407169af5a574225eccaa043b34a?context=explore) | linux/arm/v7 | 618.78 MB |
>  | [634d46e072e3](https://hub.docker.com/layers/automaticrippingmachine/automatic-ripping-machine/latest/images/sha256-634d46e072e33b0db3725aaa0ddbbfb2a97925fa5010b5861445c8876de64421?context=explore) | linux/arm64 | 671.74 MB |

But I don't see that corresponding to any update on the Repo. The last commit on the `main` branch was 4 days ago (01d0fc13d819917fea5e9c3c3636d20d35085b85), on the `devel` branch 3 weeks ago (215687d5033e1965cab87f2834f1538a96a9a631). The last commit in the arm-dependencies submodule was a day ago (automatic-ripping-machine/arm-dependencies@f949ac0c9da5dbbf3ec584dc3bcbd31842e20127), but that shouldn't be triggering a DockerHub update because the submodule wasn't updated (and doesn't fit with the "6 hours ago" anyway).

The only other layer I see on the [DockerHub page](https://hub.docker.com/layers/automaticrippingmachine/automatic-ripping-machine/latest/images/sha256-30a5f129e72f7d40916ee4b4293dcaaef66f02f4f62b198070fb3618f09b5879?context=explore) is Ubuntu itself, and that surely doesn't get an update every day. In fact the [22.04 image](https://hub.docker.com/layers/library/ubuntu/22.04/images/sha256-3d1556a8a18cf5307b121e0a98e93f1ddf1f3f8e092f1fddfd941254785b95d7?context=explore) it's pulling was:
> Last pushed
> 21 days ago by [doijanky](https://hub.docker.com/u/doijanky)

So... what gives? I don't really mind personally, it's just... a little weird.

**Comment by zultron** [comment]
https://github.com/automatic-ripping-machine/automatic-ripping-machine/actions

Check out CI.  Maybe it's publishing images on every CI run instead of only on tag released, like you expected.

(And while you're at it, check that it's not publishing Docker images for PR branches, which would be a massive security problem!)

**Reply by microtechno9000** [reply, marked as answer]
The docker image is built nightly and on any changes to main

**Reply by mihawk90** [reply]
> https://github.com/automatic-ripping-machine/automatic-ripping-machine/actions
> 
> Check out CI. Maybe it's publishing images on every CI run instead of only on tag released, like you expected.

That's what I thought and why I mentioned the last commit dates. But since the commit dates don't coincide with the DockerHub dates, there would have to be pushes outside of commits. I checked the github action files but I couldn't see where it would be pushing to DockerHub.

> check that it's not publishing Docker images for PR branches

That's sort of what I'm expecting, but I can never tell whether the container on DockerHub was actually updated or if it's just the tag being renewed.

**Reply by mihawk90** [reply]
> The docker image is built nightly

Is there any reason for that when there haven't been any changes anyway? On commits to main makes sure it's the latest code anyway, no?

**Reply by microtechno9000** [reply]
I didn't set it up, but one of the other Devs did. From what I understand it is to keep the MakeMKV beta key and application current.

Edit. There may be a better way to do this. Is there an issue that a nightly build is creating?

**Reply by mihawk90** [reply]
> MakeMKV beta key and application current.

The beta key doesn't come with the container, it's obtained anytime a rip using MakeMKV starts (it's in the logfiles). The key also seems to be removed from the settings file after each run (I assume to avoid complications due to MakeMKV starting with an old beta key).
As for the MakeMKV update that's certainly a concern but it doesn't update *that* often.

I wonder if we could utilise [Release Monitoring](https://release-monitoring.org/project/177148/) for this. I'm not sure if there's some integration with GitHub CI though.
edit: actually, maybe dependabot could do this as well? Either trigger CI directly or create a PR and merge to trigger it?

> Is there an issue that a nightly build is creating?

Not an "issue" per se, more of an annoyance. TrueNAS can send an email anytime there is an update for an app (which unfortunately can't be set per app, it's for all installed apps). So every day I get an email about an "update", I need to log into the UI, update the app - because I don't know if it's an actual update or not.
It's also a lot of wasted bandwidth, which is not so much a problem with my connection but there are probably people where a gig a day can become cumbersome.

**Reply by microtechno9000** [reply]
True, the beta key isn't in the container but a current version is needed to keep the key current.

Suggest raising an issue with details and info. The devs don't run TrueNAS, so we haven't run into this issue before.

Dependabot does a bunch already for the dependencies container and the main ARM image. Can have a look to see what it might be able to do for MakeMKV.
